src/simulator_core/solver/network/network.py

```diff
@@ -74,58 +74,77 @@ def add_existing_asset(self, asset: BaseAsset) -> uuid.UUID:
         return asset.name
 
     def connect_assets(
-        self, asset1: uuid.UUID, connection_point_1: int, asset2: uuid.UUID, connection_point_2: int
+            self, asset1_id: uuid.UUID, connection_point_1: int, asset2_id: uuid.UUID,
+            connection_point_2: int
     ) -> uuid.UUID:
         """Method to connect to assets at the given connection points.
 
         This method connects the two assets if the exists. It checks if they already are connected.
         If not a new node is created. Otherwise, both are connected to the existing node.
         The id of the node connecting the two is returned
 
-        :param asset1: id of first asset to be connected
+        :param asset1_id: id of first asset to be connected
         :param connection_point_1: Connection point of first asset to be connected
-        :param asset2: id of second asset to be connected
+        :param asset2_id: id of second asset to be connected
         :param connection_point_2: Connection point of second asset to be connected
         :return: id of node connecting the two assets
         """
-        if not self.exists_asset(asset1):
-            raise ValueError(str(asset1) + " does not exists in network")
-        elif not self.exists_asset(asset2):
-            raise ValueError(str(asset2) + " does not exists in network")
-        elif (not self.assets[asset1].is_connected(connection_point_1)) and not (
-            self.assets[asset2].is_connected(connection_point_2)
+        if not self.exists_asset(asset_id=asset1_id):
+            raise ValueError(f"{asset1_id} +  does not exists in network.")
```

Comment:
Je zou deze ValueError - of moet het een IndexError - ook kunnen raisen in exists_assets, waardoor je weer een aantal calls kwijt bent?

Reply:
```suggestion
    def _connect_both_assets():
        # both assets are not connected. Create a new node and connect everything.
        node_id = uuid.uuid4()
        self.nodes[node_id] = Node(name=asset_id)
        # Connect asset 1
        self.assets[asset1_id].connect_node(connection_point=connection_point_1,
                                            node=self.nodes[node_id])
        self.nodes[node_id].connect_asset(asset=self.assets[asset1_id],
                                            connection_point=connection_point_1)
        # Connect asset 2
        self.assets[asset2_id].connect_node(connection_point=connection_point_2,
                                            node=self.nodes[node_id])
        self.nodes[node_id].connect_asset(asset=self.assets[asset2_id],
                                            connection_point=connection_point_2)
        return node_id
```